src/backlight/backlight.c

```diff
@@ -192,11 +192,8 @@ static int validate_device(sd_device *device) {
         if (r < 0)
                 return log_device_debug_errno(device, r, "Failed to get sysname: %m");
 
-        r = sd_device_get_subsystem(device, &subsystem);
-        if (r < 0)
-                return log_device_debug_errno(device, r, "Failed to get subsystem: %m");
-        if (!streq(subsystem, "backlight"))
-                return true;
+        if (!device_in_subsystem(device, "backlight"))
```

Comment:
the new helper could be used elsewhere too. In src/login/logind-core.c i see one case where it would make sense in src/mount/mount-tool.c too, src/shared/blockdev-util.c, src/udev/udev-builtin-net_id.c